### PR TITLE
10/combat animations

### LINKED ISSUE #366: [Team 10] General Combat Animations
# **Description**
**Task:** General Combat Animations
**Feature:** Combat Enhancements: UI, Animations, Items, and Multi-Attack (#334)

This task will involve creating animations for all buttons and actions involved in combat so that each user action during combat is met with visual stimuli. This will invovle working on enemy, player, and boss animations as dependencies aside from any general animations. 

# **Dependencies**

- [ ] Boss Animations in Combat Screen (#342)
- [ ] Player Animations in the Combat Screen (#341)
- [ ] Enemy Animations in Combat Screen (#340)
 
# **Milestones**
List of steps that need to be completed for this task.
- [x] Add attacking animations for players (Sept 20.)
- [x] Add attacking animations for enemies (Sept 27.)
- [x] Add sleeping animations, if decided with team (Sept 4.)
- [x] Add guarding animations, if decided with team (Sept 4.)
 
**Completion Deadline:** Sept 4.
 
# **Documentation**

 
# **Member**
- Yessica (@yessica-s)

### PR DESCRIPTION
**Pull request by yessica-s**
# Description

This PR contains the general combat animations for both players and enemy entites. It performs player animations upon users selecting a move option and enemy animations upon update of the dialogue box to sync enemy attack text and visual animations.

Fixes / Closes #366 (General Combat Animations)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Recorded visual testing has been done to ensure the components behave as expected and integrate correctly with existing combat features. 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/components/combat/CombatManager.java [hunk 1]
```diff
@@ -273,14 +292,19 @@ private Entity getFasterEntity() {
     private void checkCombatEnd() {
         if (playerStats.getHealth() <= 0) {
             if (enemy.getComponent(CombatStatsComponent.class).isBoss()) {
+                //nullifyCombatDialogueListener();
                 this.getEntity().getEvents().trigger("combatLossBoss");
                 GameState.resetState();
                 SaveHandler.delete(GameState.class, "saves", FileLoader.Location.LOCAL);
             } else {
                 this.getEntity().getEvents().trigger("combatLoss");
+                //nullifyCombatDialogueListener();
```
Comment (MargoCode) [inline source/core/src/main/com/csse3200/game/components/combat/CombatManager.java:301]: bunch of commented out lines in this file. do u need them or can they be deleted?

File: source/core/src/main/com/csse3200/game/components/combat/CombatManager.java [hunk 2]
```diff
@@ -412,6 +440,64 @@ private void displayCombatResults() {
...
+        dialogueBoxCombatListener = new InputListener() {
+            @Override
+            public boolean touchDown(InputEvent event, float x, float y, int pointer, int button) {
+
+                combatAnimationDisplay.dispose();
+
+                Label db = ServiceLocator.getDialogueBoxService().getCurrentOverlay().getLabel();
+                String currentText = String.valueOf(db.getText());
+
+                //            TODO: replace Label code with code below due in next PR
```
Comment (MargoCode) [inline source/core/src/main/com/csse3200/game/components/combat/CombatManager.java:469]: is this still for the next pr or should this be deleted?

File: source/core/src/main/com/csse3200/game/components/combat/CombatExitDisplay.java [hunk 3]
```diff
@@ -73,6 +73,7 @@ public void changed(ChangeEvent changeEvent, Actor actor) {
               @Override
               public void changed(ChangeEvent changeEvent, Actor actor) {
           if (enemy.getComponent(CombatStatsComponent.class).isBoss()) {
+            System.out.println("here");
```
Comment (MargoCode) [inline source/core/src/main/com/csse3200/game/components/combat/CombatExitDisplay.java:76]: should it use logger instead?

File: source/core/src/main/com/csse3200/game/components/quests/AchievementDisplay.java
Comment (MargoCode) [inline source/core/src/main/com/csse3200/game/components/quests/AchievementDisplay.java:1]: good removal :)

File: source/core/src/main/com/csse3200/game/screens/CombatScreen.java [hunk 4]
```diff
@@ -101,6 +101,12 @@ public void render(float delta) {
   @Override
   public void resize(int width, int height) {
     renderer.resize(width, height);
+    // CombatArea.spawnTerrain();
```
Comment (MargoCode) [inline source/core/src/main/com/csse3200/game/screens/CombatScreen.java:104]: can this commented out code be deleted?

File: source/core/src/test/com/csse3200/game/screens/CombatScreenTest.java
Comment (MargoCode) [inline source/core/src/test/com/csse3200/game/screens/CombatScreenTest.java:1]: r u gonna complete these tests or is this for next sprint?

**Review by MargoCode — APPROVED** [review]
looks good, but is the croc supposed to have animations cuz i can't seem to find any? if that's by choice, all good except some minor things.

File: source/core/src/main/com/csse3200/game/components/combat/CombatManager.java (hunk 2 above)
Comment (yessica-s) [reply source/core/src/main/com/csse3200/game/components/combat/CombatManager.java:464]: yes next pr probably. thanks for reviewing :)

**Comment by lhick2108** [comment]
It says there is currently conflicts that have to be resolved.

**Comment by yessica-s** [comment]
> It says there is currently conflicts that have to be resolved.

my bad. fixed

File: hs_err_pid29768.log
Comment (Jul-Wong) [inline hs_err_pid29768.log:1]: I think we're supposed to delete these error logs?

File: q
Comment (Jul-Wong) [inline q:1]: I think this isn't necessary too

File: hs_err_pid29768.log
Comment (yessica-s) [reply hs_err_pid29768.log:1]: fixed tysm!

**Comment by yessica-s** [comment]
> Looks good, just try to keep those unnecessary files out of main

all fixed, thank you!

**Pull request merged**